lib/cocoapods/sandbox/path_list.rb

```diff
@@ -47,15 +50,12 @@ def read_file_system
         unless root.exist?
           raise Informative, "Attempt to read non existent folder `#{root}`."
         end
-        root_length  = root.to_s.length + 1
         escaped_root = escape_path_for_glob(root)
-        paths = Dir.glob(escaped_root + '**/*', File::FNM_DOTMATCH).sort_by(&:upcase)
-        absolute_dirs  = paths.select { |path| File.directory?(path) }
-        relative_dirs  = absolute_dirs.map  { |p| p[root_length..-1] }
-        absolute_paths = paths.reject { |p| p == "#{root}/." || p == "#{root}/.." }
-        relative_paths = absolute_paths.map { |p| p[root_length..-1] }
-        @files = relative_paths - relative_dirs
-        @dirs  = relative_dirs.map { |d| d.gsub(/\/\.\.?$/, '') }.reject { |d| d == '.' || d == '..' } .uniq
+        absolute_paths = Pathname.glob(escaped_root + '**/*', File::FNM_DOTMATCH)
+        dirs_and_files = absolute_paths.reject { |path| path.basename.to_s =~ /^\.\.?$/ }
+        relative_paths = dirs_and_files.map { |path| path.relative_path_from(root) }
+        sorted_paths = relative_paths.map(&:to_s).sort_by(&:upcase)
+        @dirs, @files = sorted_paths.partition { |path| File.directory?(root + path) }
```

Comment:
This can be `File.directory?(root + path)` or also `(root + path).directory?` if we want to stick with Pathname methods.